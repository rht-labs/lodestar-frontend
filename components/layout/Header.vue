<template>
  <header role="banner" class="pf-c-page__header">
    <div class="pf-c-page__header-brand">
      <div class="pf-c-page__header-brand-toggle">
        <button
          id="page-default-nav-example-nav-toggle"
          class="pf-c-button pf-m-plain"
          type="button"
          @click="$emit('toggle-sidebar')"
        >
          <i class="fas fa-bars" aria-hidden="true"></i>
        </button>
      </div>
      <a class="pf-c-page__header-brand-link">
        <img
          class="pf-c-brand"
          src="~assets/logo-with-labs.png"
          alt="PatternFly logo"
        />
      </a>
    </div>
    <HeaderTools />
  </header>
</template>

<script>
import HeaderTools from '~/components/layout/HeaderTools.vue'

export default {
  components: {
    HeaderTools
  }
}
</script>
