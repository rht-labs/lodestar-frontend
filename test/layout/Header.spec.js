import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import Header from '@/components/layout/Header.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
let mockStore
let wrapper

describe('Header', () => {
  beforeEach(() => {
    mockStore = new Vuex.Store({
      modules: {
        auth: {
          namespaced: false,
          state: { user: { name: 'name' } }
        }
      }
    })
  })

  test('is a Vue instance', () => {
    wrapper = shallowMount(Header, { store: mockStore, localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    wrapper = shallowMount(Header, { store: mockStore, localVue })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
