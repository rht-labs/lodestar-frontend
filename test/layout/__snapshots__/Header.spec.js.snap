// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `"<header role=\\"banner\\" class=\\"pf-c-page__header\\"><div class=\\"pf-c-page__header-brand\\"><div class=\\"pf-c-page__header-brand-toggle\\"><button id=\\"page-default-nav-example-nav-toggle\\" type=\\"button\\" class=\\"pf-c-button pf-m-plain\\"><i aria-hidden=\\"true\\" class=\\"fas fa-bars\\"></i></button></div> <a class=\\"pf-c-page__header-brand-link\\"><img src=\\"~assets/logo-with-labs.png\\" alt=\\"PatternFly logo\\" class=\\"pf-c-brand\\"></a></div> <headertools-stub></headertools-stub></header>"`;
